File: src/main/java/pl/edu/pbs/spp/Route.java
package pl.edu.pbs.spp;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class Route {

    private String from;
    private String to;
    private double cost;

    @JsonIgnore
    private int successorNode;

    private List<String> nodesPath;
    private List<String> edgesPath;
}

File: tsp-solver-frontend/src/app/model/request/spp/Route.ts
export class Route {
  from: string;
  to: string;
  cost: number;
  nodesPath: number[]
  edgesPath: number[]

  constructor(from: string, to: string, cost: number, nodesPath: number[], edgesPath: number[]) {
    this.from = from;
    this.to = to;
    this.cost = cost;
    this.nodesPath = nodesPath;
    this.edgesPath = edgesPath;
  }

}
